src/main/java/io/vertx/core/file/impl/FileSystemImpl.java

```diff
@@ -1045,11 +1045,16 @@ public boolean accept(File dir, String name) {
             } else {
               files = file.listFiles(fnFilter);
             }
-            List<String> ret = new ArrayList<>(files.length);
-            for (File f : files) {
-              ret.add(f.getCanonicalPath());
+            if (null == files || files.length == 0) {
+              return new ArrayList<>();
```

Comment:
Why not `Collections.emptyList()` ?

Reply:
Because `Collections.emptyList()` is an immutable list . Creating a new instance would be compatible with previous version

Reply:
I get the reasons, I am just thinking, shouldn't this list actually be immutable? For example, the underlying data is an array. We wouldn't be able to add entries to the array because its size is fixed. (We can mutate elements, though). So, my question was more, should we align the behavior of the underlying API with the `FileSystem` API?

WDYT @vietj ?

Reply:
I agree we need to return a new array list here, perhaps we can optimize and instead return a list with a minimal initial capacity as it should not be updated

Reply:
e.g `new ArrayList(0)` will not allocate an array